stripe/src/main/java/com/stripe/android/StripePaymentController.kt

```diff
@@ -300,6 +302,69 @@ internal class StripePaymentController internal constructor(
         stripeRepository.retrieveSource(sourceId, clientSecret, requestOptions, callback)
     }
 
+    override fun authenticateAlipay(
+        intent: StripeIntent,
+        clientSecret: String,
+        stripeAccountId: String?,
+        authenticationHandler: AlipayAuthenticationHandler,
+        callback: ApiResultCallback<PaymentIntentResult>
+    ) {
+        AlipayAuthenticationTask(
+            intent,
+            authenticationHandler,
+            object : ApiResultCallback<Int> {
+                override fun onSuccess(result: Int) {
+                    val requestOptions = ApiRequest.Options(
+                        apiKey = publishableKey,
+                        stripeAccount = stripeAccountId
+                    )
+
+                    stripeRepository.retrieveIntent(
+                        clientSecret,
+                        requestOptions,
+                        expandFields = EXPAND_PAYMENT_METHOD,
+                        callback = createPaymentIntentCallback(
+                            requestOptions, result, "", false, callback
```

Comment:
@mshafrir-stripe I'm just setting `shouldCancelSource` to false here and passing an empty string for the source id but I'm a little unclear on exactly what this is for. Does this make sense?

Reply:
I think we only need to do this for 3DS2, but we can confirm.